src/jobs/pauseAppCheck.ts

```diff
@@ -0,0 +1,165 @@
+    count,
+    unitOfTime as unitOfTime.DurationConstructor,
+  );
+  const approvalDayStart = moment(approvalDate).startOf('day').toDate();
+  // TODO: depending on how expiry/renewal is handled for applications that are never attested, will need to modify this query
+  // to check for PAUSED state and date range of attestationByUtc to expiresAtUtc
+  const query: FilterQuery<ApplicationDocument> = {
+    state: 'APPROVED',
+    approvedAtUtc: {
+      // filter for any time period equal to or past attestationByUtc in case an application that should have been paused previously
+      // is caught on a subsequent job run, as it will still be APPROVED and not have an attestedAtUtc value
+      $gte: approvalDayStart,
+    },
+    // tslint:disable-next-line:no-null-keyword
+    $or: [{ attestedAtUtc: { $exists: false } }, { attestedAtUtc: { $eq: null } }], // check the applicant has not already attested, value may be null after renewal
+  };
+
+  return query;
+};
+
+const getPausedAppsReport = async (
+  emailClient: Transporter<SMTPTransport.SentMessageInfo>,
+  user: Identity,
+  currentDate: Date,
+) => {
+  const config = await getAppConfig();
+  const query = getPauseableQuery(config, currentDate);
+  const pauseableAppCount = await ApplicationModel.find(query).countDocuments();
+  // if no applications fit the criteria, return initial report
+  if (pauseableAppCount === 0) {
+    logger.info(`${JOB_NAME} - No applications need to be paused at this time.`);
+    logger.info(`${JOB_NAME} - Generating report.`);
+    return getEmptyReport();
+  }
+  logger.info(`${JOB_NAME} - There are ${pauseableAppCount} apps that should be PAUSED.`);
+  const pauseableApps = await searchPauseableApplications(query);
+
+  const results: PromiseSettledResult<any>[][] = [];
+  const requests = chunk(pauseableApps, REQUEST_CHUNK_SIZE).map(
+    async (appChunk: ApplicationDocument[]) => {
+      return Promise.allSettled(
+        appChunk.map(async (app: ApplicationDocument) => {
+          const updatedAppObj = (await pauseApplication(
+            app,
+            user,
+            PauseReason.PENDING_ATTESTATION,
+          )) as Application;
+          if (updatedAppObj.state !== 'PAUSED') {
+            logger.error(
+              `${JOB_NAME} - Failed to transition ${updatedAppObj.appId} from ${app.state} to PAUSED state.`,
+            );
+            logger.error(
+              `${JOB_NAME} - ${updatedAppObj.appId} is in ${updatedAppObj.state} state.`,
+            );
+            throw new Error(
+              `${JOB_NAME} - Failed to transition ${updatedAppObj.appId} from ${app.state} to PAUSED state.`,
+            );
+          }
+          logger.info(
+            `${JOB_NAME} - Successfully transitioned app ${updatedAppObj.appId} to PAUSED state.`,
+          );
+          await onStateChange(updatedAppObj, app, emailClient, config);
+          return updatedAppObj;
+        }),
+      );
+    },
+  );
+
+  for (const chunk of requests) {
```

Comment:
same comment as before, we may be able to just await allResults straight from our Promise.allSettled